internal/tools/aws/database_multitenant.go

```diff
@@ -0,0 +1,717 @@
+	}
+
+	err = d.createUserIfNotExist(ctx, installationSecret.MasterUsername, installationSecret.MasterPassword)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create a Mattermost database schema for installation ID %s", d.installationID)
+	}
+
+	err = d.grantUserFullPermissions(ctx, installationDatabaseName, installationSecret.MasterUsername)
+	if err != nil {
+		return errors.Wrapf(err, "failed to grant permissions to Mattermost database user for installation ID %s", d.installationID)
+	}
+
+	databaseInstallationIDs, err := store.AddMultitenantDatabaseInstallationID(*lockedCluster.cluster.DBClusterIdentifier, d.installationID)
+	if err != nil {
+		return errors.Wrapf(err, "failed to add the installation ID %s to multitenant database ID %s", d.installationID, *lockedCluster.cluster.DBClusterIdentifier)
+	}
+
+	err = d.updateCounterTag(lockedCluster.cluster.DBClusterArn, len(databaseInstallationIDs))
+	if err != nil {
+		return errors.Wrapf(err, "failed to update tag:counter in RDS cluster ID %s", *lockedCluster.cluster.DBClusterIdentifier)
+	}
+
+	logger.Infof("Multitenant database ID %s has %d installations", *lockedCluster.cluster.DBClusterIdentifier, len(databaseInstallationIDs))
+
+	return nil
+}
+
+// Helpers
+
+type validatedAndLockedRDSCluster struct {
+	unlock  func(log.FieldLogger)
+	cluster *rds.DBCluster
+}
+
+func (d *RDSMultitenantDatabase) validateAndLockRDSCluster(multitenantDatabases []*model.MultitenantDatabase, store model.InstallationDatabaseStoreInterface, logger log.FieldLogger) (*validatedAndLockedRDSCluster, error) {
+	for _, multitenantDatabase := range multitenantDatabases {
+		installations, err := multitenantDatabase.GetInstallationIDs()
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get installations for multitenant database ID %s", multitenantDatabase.ID)
+		}
+
+		if len(installations) < DefaultRDSMultitenantDatabaseCountLimit || installations.Contains(d.installationID) {
+			unlockFn, err := d.lockMultitenantDatabase(multitenantDatabase.ID, store)
+			if err != nil {
+				logger.WithError(err).Errorf("Failed to lock multitenant database ID %s", multitenantDatabase.ID)
+				continue
+			}
+
+			multitenantDatabase, err = store.GetMultitenantDatabase(multitenantDatabase.ID)
+			if err != nil {
+				logger.WithError(err).Errorf("Failed to get multitenant database ID %s", multitenantDatabase.ID)
+				continue
+			}
+
+			err = d.validateMultitenantDatabaseInstallations(installations, multitenantDatabase)
+			if err != nil {
+				unlockFn(logger)
+				logger.WithError(err).Errorf("Multitenant database ID %s validation failed", multitenantDatabase.ID)
+				continue
+			}
+
+			rdsCluster, err := d.describeRDSCluster(multitenantDatabase.ID)
+			if err != nil {
+				unlockFn(logger)
+				logger.WithError(err).Errorf("Failed to describe the multitenant RDS cluster ID %s", multitenantDatabase.ID)
+				continue
```

Comment:
specifically the above mentioned helper method would let you turn these `continue` statements into `return`s, I think, which might make it easier to think about this routine.

Reply:
I see, disregard the question above 👍